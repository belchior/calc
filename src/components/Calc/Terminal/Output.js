import React, {Component} from 'react';

class Output extends Component {
  componentDidUpdate() {
    this.outputElem.scrollTop = this.outputElem.scrollHeight;
  }

  render() {
    const results = this.props.results || [];
    return (
      <div className="output" ref={(output) => this.outputElem = output}>
        {results.map((item, index) => <output className="result" key={index}>{item}</output>)}
      </div>
    );
  }
}

export default Output;
