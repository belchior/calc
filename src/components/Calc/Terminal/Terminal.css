/*
  Calc - Terminal

  Helper for unit convertion
  1px = 0.0625em
  2px = 0.125em
  3px = 0.1875em
  4px = 0.25em
  5px = 0.3125em
  6px = 0.375em
  7px = 0.4375em
  8px = 0.5em
  9px = 0.5625em
  10px = 0.625em
  11px = 0.6875em
  12px = 0.75em
  13px = 0.8125em
  14px = 0.875em
  15px = 0.9375em
  16px = 1em
*/
.Terminal {
  box-sizing: border-box;
  border-radius: 0.6em;
  width: 18.75em;
  font-size: 1em;
  color: #ffffff;
  outline: none;
}
.Terminal .display {
  box-sizing: border-box;
  overflow: hidden;
  border-radius: 0.6em 0.6em 0 0;
  height: 7.5em;
  line-height: 1.5em;
  background-color: #333333;
}
.Terminal .output {
  overflow: hidden;
  height: 5.625em;
}
.Terminal .output .result {
  display: inline-block;
  padding-right: 0.3125em;
  width: 100%;
  line-height: 1.4424em;
  text-align: right;
  font-size: 1.25em;
}
.Terminal .output .result:after {
  content: " <";
  margin: 0 0.3125em 0 0;
  color: #66cc66;
}
.Terminal .input {
  box-sizing: border-box;
  display: block;
  margin: 0;
  border: 0;
  padding: 0;
  padding-right: 1.5em;
  width: 100%;
  height: 1.429em;
  line-height: 1.25em;
  font-size: 1.25em;
  text-align: right;
  color: #ffffff;
  background-color: #333333;
}
.Terminal .input:after {
  content: " $";
  margin: 0 -1.1875em 0 0;
  color: #66cc66;
}
.Terminal .keyboard {
  display: table;
  border-radius: 0 0 0.6em 0.6em;
  width: 18.8em;
  background-color: #000000;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}
.Terminal .btn {
  box-sizing: border-box;
  float: left;
  display: inline-block;
  border-radius: 0.1924em;
  border: 0;
  width: 3.5715em;
  height: 3.334em;
  font-size: 1.3125em;
  line-height: 3.5715em;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  transition-property: background;
  transition-duration: 0.5s;
  transition-timing-function: ease-out;
}
.Terminal .btn:active {
  background: #333333;
}
.Terminal .result {
  cursor: pointer;
}
.Terminal .btn:focus,
.Terminal .output:focus,
.Terminal .input:focus,
.Terminal .result:focus {
  outline: none;
  box-shadow: inset 0 0 0 1px #999999;
}
