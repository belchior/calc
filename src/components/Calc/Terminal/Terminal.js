import React, {Component} from 'react';
import Output from './Output.js';
import './Terminal.css';
import {connect} from 'react-redux';

class Terminal extends Component {
  render() {
    let className = 'Terminal';
    if (this.props.error) {
      className += ' shake-horizontal';
      this.props.disableError();
    }

    let formula = this.props.formula;
    if (this.props.startNewCalc) {
      formula = '';
    }
    return (
      <form className={className}>
        <div className="display">
          <Output results={this.props.results} />
          <p className="input">{formula}</p>
        </div>
        <div className="keyboard">
          <div className="row">
            <button onClick={this.props.parenthesisLeftClick} type="button" className="btn" data-name="parenthesisOpen" data-value="(" title="parenthesis opening">(</button>
            <button onClick={this.props.parenthesisRightClick} type="button" className="btn" data-name="parenthesisClose" data-value=")" title="parenthesis closing">)</button>
            <button onClick={this.props.deleteClick} type="button" className="btn" data-name="delete" title="delete">&lt;</button>
            <button onClick={this.props.clearClick} type="button" className="btn" data-name="clear" title="clear">c</button>
          </div>
          <div className="row">
            <button onClick={this.props.numberClick} type="button" className="btn" data-name="number1" data-value="1" title="number 1">1</button>
            <button onClick={this.props.numberClick} type="button" className="btn" data-name="number2" data-value="2" title="number 2">2</button>
            <button onClick={this.props.numberClick} type="button" className="btn" data-name="number3" data-value="3" title="number 3">3</button>
            <button onClick={this.props.plusClick} type="button" className="btn" data-name="addition" data-value="+" title="plus">+</button>
          </div>
          <div className="row">
            <button onClick={this.props.numberClick} type="button" className="btn" data-name="number4" data-value="4" title="number 4">4</button>
            <button onClick={this.props.numberClick} type="button" className="btn" data-name="number5" data-value="5" title="number 5">5</button>
            <button onClick={this.props.numberClick} type="button" className="btn" data-name="number6" data-value="6" title="number 6">6</button>
            <button onClick={this.props.minusClick} type="button" className="btn" data-name="subtraction" data-value="-" title="minus">−</button>
          </div>
          <div className="row">
            <button onClick={this.props.numberClick} type="button" className="btn" data-name="number7" data-value="7" title="number 7">7</button>
            <button onClick={this.props.numberClick} type="button" className="btn" data-name="number8" data-value="8" title="number 8">8</button>
            <button onClick={this.props.numberClick} type="button" className="btn" data-name="number9" data-value="9" title="number 9">9</button>
            <button onClick={this.props.multiplicationClick} type="button" className="btn" data-name="multiplication" data-value="×" title="times">×</button>
          </div>
          <div className="row">
            <button onClick={this.props.dotClick} type="button" className="btn" data-name="dot" data-value="." title="dot">.</button>
            <button onClick={this.props.numberClick} type="button" className="btn" data-name="number0" data-value="0" title="number 0">0</button>
            <button onClick={this.props.equalsClick} type="button" className="btn" data-name="equality" title="equals">=</button>
            <button onClick={this.props.divisionClick} type="button" className="btn" data-name="division" data-value="÷" title="divided">÷</button>
          </div>
        </div>
      </form>
    );
  }
}

const mapStateToProps = (state) => {
  return state.terminal;
};

const mapDispatchToProps = (dispatch) => {
  return {
    clearClick(e) {
      dispatch({type: 'CLEAR', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    deleteClick(e) {
      dispatch({type: 'DELETE', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    divisionClick(e) {
      dispatch({type: 'DIVISION', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    dotClick(e) {
      dispatch({type: 'DOT', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    equalsClick(e) {
      dispatch({type: 'EQUALS', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    minusClick(e) {
      dispatch({type: 'MINUS', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    multiplicationClick(e) {
      dispatch({type: 'MULTIPLY', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    numberClick(e) {
      dispatch({type: 'NUMBER', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    parenthesisLeftClick(e) {
      dispatch({type: 'PARENTHESIS_LEFT', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    parenthesisRightClick(e) {
      dispatch({type: 'PARENTHESIS_RIGHT', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    plusClick(e) {
      dispatch({type: 'PLUS', calc: 'terminal', payload: e.target.getAttribute('data-value')});
    },
    disableError() {
      setTimeout(() => dispatch({type: 'ERROR', calc: 'terminal', payload: false}), 400);
    },
  };
};

export {Terminal};
export default connect(mapStateToProps, mapDispatchToProps)(Terminal);
