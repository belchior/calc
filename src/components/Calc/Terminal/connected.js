import { connect } from 'react-redux';
import Terminal from './Terminal';

export const mapStateToProps = (state) => {
  return state.terminal;
};

export const mapDispatchToProps = (dispatch) => {
  return {
    clearClick(e) {
      dispatch({ type: 'CLEAR', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    deleteClick(e) {
      dispatch({ type: 'DELETE', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    divisionClick(e) {
      dispatch({ type: 'DIVISION', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    dotClick(e) {
      dispatch({ type: 'DOT', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    equalsClick(e) {
      dispatch({ type: 'EQUALS', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    minusClick(e) {
      dispatch({ type: 'MINUS', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    multiplicationClick(e) {
      dispatch({ type: 'MULTIPLY', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    numberClick(e) {
      dispatch({ type: 'NUMBER', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    parenthesisLeftClick(e) {
      dispatch({ type: 'PARENTHESIS_LEFT', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    parenthesisRightClick(e) {
      dispatch({ type: 'PARENTHESIS_RIGHT', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    plusClick(e) {
      dispatch({ type: 'PLUS', calc: 'terminal', payload: e.target.getAttribute('data-value') });
    },
    disableError() {
      setTimeout(() => dispatch({ type: 'ERROR', calc: 'terminal', payload: false }), 400);
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Terminal);
