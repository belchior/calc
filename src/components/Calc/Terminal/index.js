import Terminal, {mapStateToProps, mapDispatchToProps} from './connected';

export {mapStateToProps, mapDispatchToProps};
export default Terminal;
