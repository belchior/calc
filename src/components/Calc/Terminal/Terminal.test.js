import React from 'react';
import Terminal from './Terminal';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Terminal', () => {
  const renderer = new ShallowRenderer();

  it('should render without crashing', () => {
    renderer.render(<Terminal />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it('should add css class shake-horizontal when error props is equal true', () => {
    const disableError = jest.fn();

    renderer.render(<Terminal error={true} disableError={disableError}/>);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it('should render the value of formula into input class', () => {
    renderer.render(<Terminal formula="123" />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it('should make element with css class input empty when startNewCalc props is true', () => {
    renderer.render(<Terminal formula="123" startNewCalc={true} />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
