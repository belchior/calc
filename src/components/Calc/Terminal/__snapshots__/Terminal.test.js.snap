// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terminal should add css class shake-horizontal when error props is equal true 1`] = `
<form
  className="Terminal shake-horizontal"
>
  <div
    className="display"
  >
    <Connect(Output)
      results={undefined}
    />
    <p
      className="input"
    />
  </div>
  <div
    className="keyboard"
  >
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="parenthesisOpen"
        data-value="("
        onClick={undefined}
        title="parenthesis opening"
        type="button"
      >
        (
      </button>
      <button
        className="btn"
        data-name="parenthesisClose"
        data-value=")"
        onClick={undefined}
        title="parenthesis closing"
        type="button"
      >
        )
      </button>
      <button
        className="btn"
        data-name="delete"
        onClick={undefined}
        title="delete"
        type="button"
      >
        &lt;
      </button>
      <button
        className="btn"
        data-name="clear"
        onClick={undefined}
        title="clear"
        type="button"
      >
        c
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number1"
        data-value="1"
        onClick={undefined}
        title="number 1"
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        data-name="number2"
        data-value="2"
        onClick={undefined}
        title="number 2"
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        data-name="number3"
        data-value="3"
        onClick={undefined}
        title="number 3"
        type="button"
      >
        3
      </button>
      <button
        className="btn"
        data-name="addition"
        data-value="+"
        onClick={undefined}
        title="plus"
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number4"
        data-value="4"
        onClick={undefined}
        title="number 4"
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        data-name="number5"
        data-value="5"
        onClick={undefined}
        title="number 5"
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        data-name="number6"
        data-value="6"
        onClick={undefined}
        title="number 6"
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        data-name="subtraction"
        data-value="-"
        onClick={undefined}
        title="minus"
        type="button"
      >
        −
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number7"
        data-value="7"
        onClick={undefined}
        title="number 7"
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        data-name="number8"
        data-value="8"
        onClick={undefined}
        title="number 8"
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        data-name="number9"
        data-value="9"
        onClick={undefined}
        title="number 9"
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        data-name="multiplication"
        data-value="×"
        onClick={undefined}
        title="times"
        type="button"
      >
        ×
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="dot"
        data-value="."
        onClick={undefined}
        title="dot"
        type="button"
      >
        .
      </button>
      <button
        className="btn"
        data-name="number0"
        data-value="0"
        onClick={undefined}
        title="number 0"
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        data-name="equality"
        onClick={undefined}
        title="equals"
        type="button"
      >
        =
      </button>
      <button
        className="btn"
        data-name="division"
        data-value="÷"
        onClick={undefined}
        title="divided"
        type="button"
      >
        ÷
      </button>
    </div>
  </div>
</form>
`;

exports[`Terminal should make element with css class input empty when startNewCalc props is true 1`] = `
<form
  className="Terminal"
>
  <div
    className="display"
  >
    <Connect(Output)
      results={undefined}
    />
    <p
      className="input"
    />
  </div>
  <div
    className="keyboard"
  >
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="parenthesisOpen"
        data-value="("
        onClick={undefined}
        title="parenthesis opening"
        type="button"
      >
        (
      </button>
      <button
        className="btn"
        data-name="parenthesisClose"
        data-value=")"
        onClick={undefined}
        title="parenthesis closing"
        type="button"
      >
        )
      </button>
      <button
        className="btn"
        data-name="delete"
        onClick={undefined}
        title="delete"
        type="button"
      >
        &lt;
      </button>
      <button
        className="btn"
        data-name="clear"
        onClick={undefined}
        title="clear"
        type="button"
      >
        c
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number1"
        data-value="1"
        onClick={undefined}
        title="number 1"
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        data-name="number2"
        data-value="2"
        onClick={undefined}
        title="number 2"
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        data-name="number3"
        data-value="3"
        onClick={undefined}
        title="number 3"
        type="button"
      >
        3
      </button>
      <button
        className="btn"
        data-name="addition"
        data-value="+"
        onClick={undefined}
        title="plus"
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number4"
        data-value="4"
        onClick={undefined}
        title="number 4"
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        data-name="number5"
        data-value="5"
        onClick={undefined}
        title="number 5"
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        data-name="number6"
        data-value="6"
        onClick={undefined}
        title="number 6"
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        data-name="subtraction"
        data-value="-"
        onClick={undefined}
        title="minus"
        type="button"
      >
        −
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number7"
        data-value="7"
        onClick={undefined}
        title="number 7"
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        data-name="number8"
        data-value="8"
        onClick={undefined}
        title="number 8"
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        data-name="number9"
        data-value="9"
        onClick={undefined}
        title="number 9"
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        data-name="multiplication"
        data-value="×"
        onClick={undefined}
        title="times"
        type="button"
      >
        ×
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="dot"
        data-value="."
        onClick={undefined}
        title="dot"
        type="button"
      >
        .
      </button>
      <button
        className="btn"
        data-name="number0"
        data-value="0"
        onClick={undefined}
        title="number 0"
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        data-name="equality"
        onClick={undefined}
        title="equals"
        type="button"
      >
        =
      </button>
      <button
        className="btn"
        data-name="division"
        data-value="÷"
        onClick={undefined}
        title="divided"
        type="button"
      >
        ÷
      </button>
    </div>
  </div>
</form>
`;

exports[`Terminal should render the value of formula into input class 1`] = `
<form
  className="Terminal"
>
  <div
    className="display"
  >
    <Connect(Output)
      results={undefined}
    />
    <p
      className="input"
    >
      123
    </p>
  </div>
  <div
    className="keyboard"
  >
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="parenthesisOpen"
        data-value="("
        onClick={undefined}
        title="parenthesis opening"
        type="button"
      >
        (
      </button>
      <button
        className="btn"
        data-name="parenthesisClose"
        data-value=")"
        onClick={undefined}
        title="parenthesis closing"
        type="button"
      >
        )
      </button>
      <button
        className="btn"
        data-name="delete"
        onClick={undefined}
        title="delete"
        type="button"
      >
        &lt;
      </button>
      <button
        className="btn"
        data-name="clear"
        onClick={undefined}
        title="clear"
        type="button"
      >
        c
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number1"
        data-value="1"
        onClick={undefined}
        title="number 1"
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        data-name="number2"
        data-value="2"
        onClick={undefined}
        title="number 2"
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        data-name="number3"
        data-value="3"
        onClick={undefined}
        title="number 3"
        type="button"
      >
        3
      </button>
      <button
        className="btn"
        data-name="addition"
        data-value="+"
        onClick={undefined}
        title="plus"
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number4"
        data-value="4"
        onClick={undefined}
        title="number 4"
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        data-name="number5"
        data-value="5"
        onClick={undefined}
        title="number 5"
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        data-name="number6"
        data-value="6"
        onClick={undefined}
        title="number 6"
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        data-name="subtraction"
        data-value="-"
        onClick={undefined}
        title="minus"
        type="button"
      >
        −
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number7"
        data-value="7"
        onClick={undefined}
        title="number 7"
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        data-name="number8"
        data-value="8"
        onClick={undefined}
        title="number 8"
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        data-name="number9"
        data-value="9"
        onClick={undefined}
        title="number 9"
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        data-name="multiplication"
        data-value="×"
        onClick={undefined}
        title="times"
        type="button"
      >
        ×
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="dot"
        data-value="."
        onClick={undefined}
        title="dot"
        type="button"
      >
        .
      </button>
      <button
        className="btn"
        data-name="number0"
        data-value="0"
        onClick={undefined}
        title="number 0"
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        data-name="equality"
        onClick={undefined}
        title="equals"
        type="button"
      >
        =
      </button>
      <button
        className="btn"
        data-name="division"
        data-value="÷"
        onClick={undefined}
        title="divided"
        type="button"
      >
        ÷
      </button>
    </div>
  </div>
</form>
`;

exports[`Terminal should render without crashing 1`] = `
<form
  className="Terminal"
>
  <div
    className="display"
  >
    <Connect(Output)
      results={undefined}
    />
    <p
      className="input"
    />
  </div>
  <div
    className="keyboard"
  >
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="parenthesisOpen"
        data-value="("
        onClick={undefined}
        title="parenthesis opening"
        type="button"
      >
        (
      </button>
      <button
        className="btn"
        data-name="parenthesisClose"
        data-value=")"
        onClick={undefined}
        title="parenthesis closing"
        type="button"
      >
        )
      </button>
      <button
        className="btn"
        data-name="delete"
        onClick={undefined}
        title="delete"
        type="button"
      >
        &lt;
      </button>
      <button
        className="btn"
        data-name="clear"
        onClick={undefined}
        title="clear"
        type="button"
      >
        c
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number1"
        data-value="1"
        onClick={undefined}
        title="number 1"
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        data-name="number2"
        data-value="2"
        onClick={undefined}
        title="number 2"
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        data-name="number3"
        data-value="3"
        onClick={undefined}
        title="number 3"
        type="button"
      >
        3
      </button>
      <button
        className="btn"
        data-name="addition"
        data-value="+"
        onClick={undefined}
        title="plus"
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number4"
        data-value="4"
        onClick={undefined}
        title="number 4"
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        data-name="number5"
        data-value="5"
        onClick={undefined}
        title="number 5"
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        data-name="number6"
        data-value="6"
        onClick={undefined}
        title="number 6"
        type="button"
      >
        6
      </button>
      <button
        className="btn"
        data-name="subtraction"
        data-value="-"
        onClick={undefined}
        title="minus"
        type="button"
      >
        −
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="number7"
        data-value="7"
        onClick={undefined}
        title="number 7"
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        data-name="number8"
        data-value="8"
        onClick={undefined}
        title="number 8"
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        data-name="number9"
        data-value="9"
        onClick={undefined}
        title="number 9"
        type="button"
      >
        9
      </button>
      <button
        className="btn"
        data-name="multiplication"
        data-value="×"
        onClick={undefined}
        title="times"
        type="button"
      >
        ×
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn"
        data-name="dot"
        data-value="."
        onClick={undefined}
        title="dot"
        type="button"
      >
        .
      </button>
      <button
        className="btn"
        data-name="number0"
        data-value="0"
        onClick={undefined}
        title="number 0"
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        data-name="equality"
        onClick={undefined}
        title="equals"
        type="button"
      >
        =
      </button>
      <button
        className="btn"
        data-name="division"
        data-value="÷"
        onClick={undefined}
        title="divided"
        type="button"
      >
        ÷
      </button>
    </div>
  </div>
</form>
`;
